lib/sudo-exec/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+use std::{
+    ffi::c_int,
+    io,
+    os::unix::process::CommandExt,
+    process::{Command, ExitStatus, Stdio},
+};
+
+use libc::getpgid;
+use signal_hook::{
+    consts::*,
+    iterator::{exfiltrator::WithOrigin, SignalsInfo},
+    low_level::siginfo::{Cause, Origin, Sent},
+};
+
+use sudo_common::context::Context;
+
+/// FIXME: are we missing any signals?
+const SIGNALS: &[c_int] = &[
+    SIGABRT, SIGALRM, SIGBUS, SIGCHLD, SIGCONT, SIGHUP, SIGINT, SIGPIPE, SIGPROF, SIGQUIT, SIGSYS,
```

Comment:
I did see some in `signal(7)` that aren't in this list:

* Some that I don't think we can or should handle (but maybe mention them in comments): `SIGEMT`, `SIGFPE`, `SIGILL`, `SIGSEGV`
* Some unused ones (but let's just ignore them): `SIGLOST`, `SIGSTKFLT`
* `SIGIO`
* `SIGPWR`
* `SIGSTOP`

Reply:
I documented why we shouldn't intercept `SIGKILL`, `SIGSTOP`, `SIGILL`, `SIGFPE` and `SIGSEGV`.

`SIGEMT`, `SIGLOST` and `SIGPWR` are not exposed as constants in `signal-hook` and I wonder why.

`SIGIO` was available but I forgot to add it :sweat_smile:.

We should try an add as many signals as possible otherwise sudo won't be able to intercept and redirect them to the child process.